types/legacy/params.go

```diff
@@ -0,0 +1,45 @@
+// Package legacy contains types and interfaces that have support removed, but may need to be exported for legacy
```

Comment:
While x/params is indeed removed from main, its tag remains an is still importable. When do you foresee the need for this?

Reply:
This ties the types to the actual `sdk` import as opposed to some other package.  At some point there will be a drift between the version of SDK used in an old tag and what a user is using

Reply:
See https://github.com/cosmos/cosmos-sdk/pull/23543/files#r1934276529